docs/damage.md

```diff
@@ -0,0 +1,29 @@
+# Damage
+
+Damage refers to the reduction of health points caused by the affected entity itself or a third party.
+Self-inflicted causes include, for instance, falling, drowning, or poison damage.
+Damage inflicted by a third party typically involves some form of hitting the affected entity.
```

Comment:
I'm not sure about this separation :thinking: I wouldn't have thought of "poison damage" as self-inflicted, for example. 

Why do we think this distinction is important? 

Ideally, the _instigator_ (I guess this is what this is technically about) could carry information about the body of water the player drowned in, the exact poison or disease they are infected with, etc. The instigator might not be another player or NPC, but it could be any _entity_ in the game.

Reply:
I do vaguely recall this from the past - poison is "internal" damage in a sense, unlike third party direct damage. Probably "self-inflicted" is a poor term for that. It would be nice if an internal damage effect had an _original_ trigger source that could be referenced as well.

> Player X died from poisoning, after getting too close to a Thornwood Cactus!

That might be a bit of a lofty goal for this point in time tho :-)

Reply:
> Why do we think this distinction is important?

This distinction is important because the directional damage indicator depends on having a direction to indicate. If we as the player are drowning in water, suffocating due to gas, or dying to poison in our system, there is no specific direction for the damage indicator to indicate as the "instigator" is in ourselves (poison) or all around us (water/gas).
We can also make the distinction between an instigator in or all around us vs an instigator at a specific location, however I have even less of a clue how to name _that_ then...

> Probably "self-inflicted" is a poor term for that.

Yeah, naming things is not exactly a strength of mine... I'm happy about any suggestions!

> It would be nice if an internal damage effect had an original trigger source that could be referenced as well.

In terms of the directional damage indicator, I'm not sure how it would help the player to know that the cactus they originally got poisoned by, for instance is somewhere to their right...

For the player the interesting point is "why  / where from do I lose health right now?" The directional damage indicator shows them, where their attacker can be found, for instance an archer. It's important for the player to understand whether there's an attacker they might be able to defend themselves against or whether the damage is caused by an "environmental" (again maybe not the right term) effect like water, gas, poison, ...